imapserver/imapmemserver/mailbox.go

```diff
@@ -156,6 +161,9 @@ func (mbox *Mailbox) appendBytes(buf []byte, options *imap.AppendOptions) *imap.
 	msg.uid = mbox.uidNext
 	mbox.uidNext++
 
+	msg.modSeq = mbox.highestModSeq
+	mbox.highestModSeq++
```

Comment:
Shouldn't we bump `mbox.highestModSeq` before setting the message's mod seq?

If we bump `mbox.highestModSeq` after, then it's not the highest mod seq, it's the next mod seq.